### DISCUSSION TITLE
Inout syntax

### DISCUSSION THREAD

**Discussion by mark-koch** [Quantinuum/guppylang#359] [General]
@acl-cqc has suggested some alternatives to the `@inout` syntax from #311 here https://github.com/CQCL/guppylang/pull/316#discussion_r1697313923:

> However, have you thought about the "pythonic" alternative, which I believe would be `def foo(x: Annotated[int, "inout"]`?`@inout` is a bastardization of python syntax anyway so what would users think? I admit `Annotated` is rather long - what about `def foo(x: Inout[int])` ? (No way, I hear you say - and I see the downside - maybe that lengthy Annotated isn't sooo bad, then....)
> 
> Not seriously recommending you change, but how different / how much simpler might that make things, would it impact the design at all?

Thoughts?

**Comment by croyzor** [comment]
Personally, I think `@inout` is fine, and the best of the options presented

**Reply by croyzor** [reply]
Guppy is not python!

**Comment by ss2165** [comment]
Closing as stale.
